Switch cri-o to system proxy configuration

dannysauer opened the pull request:
# Describe your changes
Update references to cri-o configuration to reflect use of system-wide proxy configuration.

Why is this "WIP/needs-review"?  I'm not sure how to link to the SLES admin guide proxy section.  I think one of these might be right:
https://documentation.suse.com/sles/15-SP1/single-html/SLES-admin/#id-1.3.3.6.13.6.24
https://www.suse.com/support/kb/doc/?id=7006845
Maybe also https://documentation.suse.com/sles/15-SP1/single-html/SLES-autoyast/#id-1.7.5.17.14.4 in reference to configuring the vars at SLE install time via AutoYaST?  Hopefully you have some thoughts, @r0ckarong :)

I also think the example config should be either changed or removed.  The use of a CIDR-style network range generally doesn't work for generic applications in no_proxy, but "might" work for cri-o specifically; I'm in the process of tracking down libraries in cri-o which actually do the work to see.  This (altering the example config) might be something to tie into the SUSE/avant-garde#600 rather than this change, though.

# Related Issues / Projects
Please provide links to Bugzilla and other GitHub projects with your description if they are related to the changes.

Relates to:
SUSE/avant-garde#997
SUSE/avant-garde#1247

# Enable maintainer updates
Please enable maintainer updates so we can push commits into your branch to make collaboration and reviews easier.

# Do not force push your branch
Please avoid force pushing to branches that are subject of pull requests. Force pushing breaks maintainer commits in many cases and is very hard (if not impossible) to untangle for backporting.

# Labels
Please set any (and all) appropriate labels that describe the status of the PR.

| **Label** | **Description** |
| --- | --- |
| P1 | PR should be worked on and merged as soon as possible |
| Blocked | Work can not proceed because other work has not been completed, PR can not be merged (code has not been merged but documentation is ready) |
| On-Hold | Underlying work is completed but the PR should not be merged |
| ReleaseNotes | User interaction is required after the introduction of this change and the change must be mentioned in the release notes |
| v3/v4/v4.x | Which version of the release the PR should be merged into, this can be multiple versions, please set the "Backport" label if it needs to go into a previous release |
| Needs Review | Some details of the PR are known to be incomplete and must be discussed with other engineers before merging (if possible assign reviewers or cc mention in comments), PR can not be merged |

r0ckarong commented on adoc/deployment-preparation.adoc:
```diff
@@ -172,8 +172,9 @@ as various default configurations and examples.
 
-Sometimes, you need a proxy server to be able to connects the SUSE Customer Center.
-To do that:
+Sometimes, you need a proxy server to be able to connect to the SUSE Customer Center.
```
You can replace `SUSE Customer Center` with the `{scc}` entity.

dannysauer replied:
```suggestion
Sometimes, you need a proxy server to be able to connect to the {scc}.
```
Like that?

r0ckarong commented:
@dannysauer This is set to WIP because there are more commits coming right?

dannysauer commented on adoc/admin-crio-proxy.adoc:
```diff
@@ -1,13 +1,12 @@
-container images. To configure this for {crio} you must modify the file
-`/etc/sysconfig/crio`.
+container images. The {crio} runtime uses the system proxy configuration,
+defined at `/etc/sysconfig/proxy`.  This file can be edited by hand, or via
```
```suggestion
defined at `/etc/sysconfig/proxy`. This file can be edited by hand, or via
```

dannysauer commented on adoc/admin-crio-proxy.adoc:
```diff
@@ -1,13 +1,12 @@
-`/etc/sysconfig/crio`.
+container images. The {crio} runtime uses the system proxy configuration,
+defined at `/etc/sysconfig/proxy`.  This file can be edited by hand, or via
+YaST.  For detailed instructions, see <insert link to SLES 15.1 docs and maybe
```
```suggestion
YaST. For detailed instructions, see <insert link to SLES 15.1 docs and maybe
```

dannysauer commented on adoc/admin-crio-proxy.adoc:
```diff
@@ -1,13 +1,12 @@
+container images. The {crio} runtime uses the system proxy configuration,
+defined at `/etc/sysconfig/proxy`. This file can be edited by hand, or via
+YaST. For detailed instructions, see <insert link to SLES 15.1 docs and maybe
+that knowledge base item here>.
```
I don't know what to do about a link in here.  There are a couple of potential link targets in the PR opening comment, but I don't know what the expected mechanism might be for linking back to SLES docs.  Or if we should do that at all.  Help, @r0ckarong :)

r0ckarong replied:
If you have multiple links, just add multiple links as a bulleted list. Just use the link to SLE 15 SP1 docs as you find them on documentation.suse.com. Right now we do not have a good mechanism to cross reference documents that are not "our own" (anything outside CaaSP). External linking is fine. If this gets updated in the future the link will be adjusted.

dannysauer commented:
I added the external links for AutoYaST and manual configuration; I couldn't find a useful link for specifically configuring the proxy via yast.  I also removed the unnecessary config example. 

I did add a list of acceptable value types in the no_proxy variable.  I know very little about asciidoc, so I'm not positive that I did the list properly, nor am I confident that the wording of the links is exactly what we'd like.  I'm a little concerned that maybe SUSE or yast or the knowledge base should be replaced with a variable?  I'll be online for a short while in the middle of your day if you need me to change something, @r0ckarong, or you're welcome to alter this however you see fit. :)

r0ckarong commented:
> I added the external links for AutoYaST and manual configuration; I couldn't find a useful link for specifically configuring the proxy via yast. I also removed the unnecessary config example.
> 
> I did add a list of acceptable value types in the no_proxy variable. I know very little about asciidoc, so I'm not positive that I did the list properly, nor am I confident that the wording of the links is exactly what we'd like. I'm a little concerned that maybe SUSE or yast or the knowledge base should be replaced with a variable? I'll be online for a short while in the middle of your day if you need me to change something, @r0ckarong, or you're welcome to alter this however you see fit. :)

This is all part of the review. I will amend changes and let you know.
